publishing-sdk/src/main/java/com/ably/tracking/publisher/CorePublisher.kt

```diff
@@ -54,6 +55,18 @@ internal fun createCorePublisher(
     return DefaultCorePublisher(ably, mapbox, resolutionPolicyFactory, routingProfile, logHandler)
 }
 
+/**
+ * This is a private static single thread dispatcher that will be used for all the [Publisher] instances.
+ * To assure that we process our events in the FIFO order we need to use a single threaded dispatcher.
+ * Because of that, all calls to the [launch] will execute in the order of invocation. Because threads
+ * are expensive resources we don't want to create a separate one for each instance. Therefore, we
+ * have one dispatcher shared across all [Publisher] instances. In an edge case scenario where multiple
+ * instances are active at the same time, this may lead to some slowdowns, as all the work performed
```

Comment:
While I appreciate your reasons for wanting to add this particular point to the commentary, I think we should keep as far away from premature optimisation in comments also. 😉 

```suggestion
 * instances are active at the same time, this could theoretically lead to some slowdowns, as all the work performed
```

Reply:
https://github.com/ably/ably-asset-tracking-android/pull/426/commits/9ea1ef6e643acd0fa35834c760d7235ca5783d75